src/wp-includes/blocks/comments.php

```diff
@@ -0,0 +1,220 @@
+ * @param string   $content    Block default content.
+ * @param WP_Block $block      Block instance.
+ * @return string Returns the filtered post comments for the current post wrapped inside "p" tags.
+ */
+function render_block_core_comments( $attributes, $content, $block ) {
+	global $post;
+
+	$post_id = $block->context['postId'];
+	if ( ! isset( $post_id ) ) {
+		return '';
+	}
+
+	$comment_args = array(
+		'post_id' => $post_id,
+		'count'   => true,
+		'status'  => 'approve',
+	);
+	// Return early if there are no comments and comments are closed.
+	if ( ! comments_open( $post_id ) && get_comments( $comment_args ) === 0 ) {
+		return '';
+	}
+
+	// If this isn't the legacy block, we need to render the static version of this block.
+	$is_legacy = 'core/post-comments' === $block->name || ! empty( $attributes['legacy'] );
+	if ( ! $is_legacy ) {
+		return $block->render( array( 'dynamic' => false ) );
+	}
+
+	$post_before = $post;
+	$post        = get_post( $post_id );
+	setup_postdata( $post );
+
+	ob_start();
+
+	/*
+	 * There's a deprecation warning generated by WP Core.
+	 * Ideally this deprecation is removed from Core.
+	 * In the meantime, this removes it from the output.
+	 */
+	add_filter( 'deprecated_file_trigger_error', '__return_false' );
+	comments_template();
+	remove_filter( 'deprecated_file_trigger_error', '__return_false' );
+
+	$output = ob_get_clean();
+	$post   = $post_before;
+
+	$classnames = array();
+	// Adds the old class name for styles' backwards compatibility.
+	if ( isset( $attributes['legacy'] ) ) {
+		$classnames[] = 'wp-block-post-comments';
+	}
+	if ( isset( $attributes['textAlign'] ) ) {
+		$classnames[] = 'has-text-align-' . $attributes['textAlign'];
+	}
+
+	$wrapper_attributes = get_block_wrapper_attributes(
+		array( 'class' => implode( ' ', $classnames ) )
+	);
+
+	/*
+	 * Enqueues scripts and styles required only for the legacy version. That is
+	 * why they are not defined in `block.json`.
+	 */
+	wp_enqueue_script( 'comment-reply' );
+	enqueue_legacy_post_comments_block_styles( $block->name );
+
+	return sprintf( '<div %1$s>%2$s</div>', $wrapper_attributes, $output );
+}
+
+/**
+ * Registers the `core/comments` block on the server.
+ */
+function register_block_core_comments() {
+	register_block_type_from_metadata(
+		__DIR__ . '/comments',
+		array(
+			'render_callback'   => 'render_block_core_comments',
+			'skip_inner_blocks' => true,
+		)
+	);
+}
+add_action( 'init', 'register_block_core_comments' );
+
+/**
+ * Use the button block classes for the form-submit button.
+ *
+ * @param array $fields The default comment form arguments.
+ *
+ * @return array Returns the modified fields.
+ */
+function comments_block_form_defaults( $fields ) {
+	if ( wp_is_block_theme() ) {
+		$fields['submit_button'] = '<input name="%1$s" type="submit" id="%2$s" class="%3$s wp-block-button__link ' . WP_Theme_JSON_Gutenberg::get_element_class_name( 'button' ) . '" value="%4$s" />';
```

Comment:
`WP_Theme_JSON_Gutenberg::get_element_class_name` was introduced in GB's [`lib/compat/wordpress-6.1`](https://github.com/WordPress/gutenberg/blob/132b86cd48e4f785a249175476302c66dc95ed2c/lib/compat/wordpress-6.1/class-wp-theme-json-6-1.php#L118-L129).

Reply:
Same as in https://github.com/WordPress/wordpress-develop/pull/3154#discussion_r962079543